### ISSUE TITLE
Multiple recompositions of ExposedDropdownMenuBox while scrolling

### ISSUE THREAD

**Issue by Zenturas** [JetBrains/compose-multiplatform#4364]
**Describe the bug**
Multiple recompositions of ExposedDropdownMenuBox while scrolling. Tested on LazyColumn

**Affected platforms**
Select one of the platforms below:
- Desktop

**Versions**
- Kotlin version*: 1.9.22
- Compose Multiplatform version*: 1.6.0-rc03
- OS version(s)* (required for Desktop and iOS issues): win11
- OS architecture (x86 or arm64): x64
- JDK (for desktop issues): jdk20

**To Reproduce**
run the code below, scroll up and down: 

```

import androidx.compose.foundation.border
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LazyColumnExposedDropdownMenuBoxMultipleRecompositionsBug() {
	var menuOpen by remember { mutableStateOf(false) }
	LazyColumn {
		item {
			println("exposeddropdownmenubox item recomposed")
			ExposedDropdownMenuBox(
				expanded = menuOpen,
				onExpandedChange = { menuOpen = it },
				content = {
					println("exposeddropdownmenubox recomposed")
					OutlinedTextField(
						value = "default",
						onValueChange = {},
						modifier = Modifier
							.fillMaxWidth()
							.menuAnchor(),
						readOnly = true,
						trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = menuOpen) },
					)
					ExposedDropdownMenu(
						expanded = menuOpen,
						onDismissRequest = { menuOpen = false },
						modifier = Modifier.exposedDropdownSize(),
						content = {
							repeat(10) {
								DropdownMenuItem(
									onClick = { menuOpen = false },
									text = { Text(it.toString()) }
								)
							}
						}
					)
				}
			)
		}

		item {
			println("big text item recomposed")
			Text("big text", modifier = Modifier.fillMaxWidth().height(800.dp).border(1.dp, Color.Red))
		}
	}
}

```

**Expected behavior**
same as in android - no redundant recompositions

**Additional context**
in android scrolled once down and once up

> exposeddropdownmenubox item recomposed
> exposeddropdownmenubox recomposed
> big text item recomposed
> exposeddropdownmenubox item recomposed
> exposeddropdownmenubox recomposed

in windows scrolled once down and once up

> exposeddropdownmenubox item recomposed
> exposeddropdownmenubox recomposed
> big text item recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox item recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed
> exposeddropdownmenubox recomposed

**Comment by Zenturas** [comment]
maybe anyone with windows can confirm. I found out if you remove .menuAnchor() there are no recompositions but also cant open the dropdownmenu.

**Comment by igordmn** [comment]
Reproduced on Windows.

The difference between macOs and Windows, that Windows enables smooth mouse scrolling.

If it is disabled, the issue isn't reproducible:
```
System.setProperty("compose.scrolling.smooth.enabled", "false")
```

**Comment by m-sasha** [comment]
@Zenturas Why do you expect recompositions to not occur here? As you scroll down, the first item of the `LazyColumn` goes out of the view and is removed from the composition. When you scroll back up, the item enters the composition again and is (re)composed.

The same thing happens in Android:


https://github.com/JetBrains/compose-multiplatform/assets/5230206/e7d2cb88-11cb-44ac-9288-fdd27af14bae

**Comment by m-sasha** [comment]
Ah, I see. It should recompose only once after re-entering the composition.

**Comment by m-sasha** [comment]
This was fixed in Android in 25d25eee042cd15f858180e6d4b592b2c95add7c

**Comment by m-sasha** [comment]
https://issuetracker.google.com/issues/256905251

**Comment by Zenturas** [comment]
@igordmn Unfortunately, the issue persists for me even with `System.setProperty("compose.scrolling.smooth.enabled", "false")` set.

@m-sasha Shouldn't 1.2.0-alpha05 already be integrated in version rc03 because it supports version 1.2.0?

**Comment by m-sasha** [comment]
> Shouldn't 1.2.0-alpha05 already be integrated in version rc03 because it supports version 1.2.0?

No, that commit isn't merged into our latest release.

**Comment by m-sasha** [comment]
@igordmn I can try cherry-picking that commit if you think it's urgent, or we can wait for the next upstream merge.

**Comment by Zenturas** [comment]
Is there a way to find out up to  which update has been fully merged?

**Comment by m-sasha** [comment]
It turns out the committee was merged, but it only fixed the Android implementation. I've submitted a PR to fix this on the multiplatform side.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4364">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
